pkgs/applications/audio/alsa-scarlett-gui/default.nix

```diff
@@ -24,6 +24,9 @@ stdenv.mkDerivation rec {
   sourceRoot = "source/src";
   nativeBuildInputs = [ pkg-config wrapGAppsHook4 ];
   buildInputs = [ gtk4 alsa-lib ];
+  postInstall = ''
+      sed -Ei "s/(Exec\s?=\s?).*/\1alsa-scarlett-gui\nTryExec=alsa-scarlett-gui/" $out/share/applications/vu.b4.${pname}.desktop
```

Comment:
From what I remember, can't find the exact comment in one of my MR/or where I participated substituteInPlace is preferred over sed, but will leave it to the merger if it's needed as I don't have merge permissions ✌🏻

Reply:
Just to give my 2cents to what speaks for sed over substituteInPlace (as far a I could investigate):

First I think it is not that easily testable. Which you really need for non trivial stuff. sed is a standard unix tool which you can easily test for the correct result on every unix/linux command line.

Secondly you could only use substituteInPlace on the exact match of the entire line (as I could not find any documentation about pattern matching)  which means worst case you would have to change it in the default.nix on every release. sed matches the Line from "Exec=" to the end of the line (not matter what comes after  "Exec=") and replaces that. That should make it more resistant to changes later on.

Reply:
sed also does nothing if it matches nothing unlike substituteInPlace which logs a warning.

Anyway, since we are patching multiple lines it would be easier to follow the change, if it would be a patch

Reply:
btw. I don't think try-exec is really needed as we discarded the %F stuff etc.

Reply:
Can we remove it then?

Reply:
I will test if the tryexec key can be removed and report back.

Just to summarize the most preferred way would be to use substituteInPlace to substitute the line with the Exec key exactly and replace that with the desired line?
And how would I go about applying the changes? A new PR?